src/oidcendpoint/authz/__init__.py

```diff
@@ -1,63 +1,101 @@
+        else:
+            _usage_rules = {}
 
-    def set(self, uid, client_id, permission):
         try:
-            self.permdb[uid][client_id] = permission
+            _per_client = self.endpoint_context.cdb[client_id]["token_usage_rules"]
         except KeyError:
-            self.permdb[uid] = {client_id: permission}
-
-    def permissions(self, cookie=None, **kwargs):
-        if cookie is None:
-            return None
+            pass
         else:
-            logger.debug("kwargs: %s" % sanitize(kwargs))
-
-            val = self.cookie_dealer.get_cookie_value(cookie)
-            if val is None:
-                return None
+            if _usage_rules:
+                for _token_type, _rule in _usage_rules.items():
+                    _pc = _per_client.get(_token_type)
+                    if _pc:
+                        _rule.update(_pc)
+                for _token_type, _rule in _per_client.items():
+                    if _token_type not in _usage_rules:
+                        _usage_rules[_token_type] = _rule
             else:
-                b64, _ts, typ = val
+                _usage_rules = _per_client
 
-            info = cookie_value(b64)
-            return self.get(info["sub"], info["client_id"])
+        return _usage_rules
 
-    def get(self, uid, client_id):
+    def usage_rules_for(self, client_id, token_type):
+        _token_usage = self.usage_rules(client_id=client_id)
         try:
-            return self.permdb[uid][client_id]
+            return _token_usage[token_type]
         except KeyError:
-            return None
+            return {}
 
+    def __call__(self, session_id: str, request: Union[dict, Message],
+                 resources: Optional[list] = None) -> Grant:
+        args = self.grant_config.copy()
 
-class Implicit(AuthzHandling):
-    def __init__(self, endpoint_context, permission="implicit"):
-        AuthzHandling.__init__(self, endpoint_context)
-        self.permission = permission
+        scope = request.get("scope")
+        if scope:
+            args["scope"] = scope
+
+        claims = request.get("claims")
+        if claims:
+            args["claims"] = claims.to_dict()
 
-    def permissions(self, cookie=None, **kwargs):
-        return self.permission
+        session_info = self.endpoint_context.session_manager.get_session_info(
+            session_id=session_id, grant=True
+        )
+        grant = session_info["grant"]
 
-    def get(self, uid, client_id):
-        return self.permission
+        for key, val in args.items():
+            if key == "expires_in":
+                grant.set_expires_at(val)
+            else:
+                setattr(grant, key, val)
+
+        if resources is None:
+            grant.resources = [session_info["client_id"]]
+        else:
+            grant.resources = resources
+
+        # This is where user consent should be handled
+        for interface in ["userinfo", "introspection", "id_token", "token"]:
```

Comment:
I think we should rename `token` to `access_token`, for clarity and backwards compatibility. If we leave it like this then any user, that uses claims_per_client for the access token, will have to change all his client metadata from `access_token_claims` to `token_claims`.

I don't see any reason to change the name

Reply:
OK

Reply:
There are other places where there might be reason to change 'token' to 'access_token'.